jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/AliasEndpoint.java

```diff
@@ -594,4 +594,19 @@ public Response deleteDestructive(@PathParam("id") String id, @Context HttpServl
 			return appendAllowOriginHeader(Response.status(Status.INTERNAL_SERVER_ERROR), request);
 		}
 	}
+
+	@POST
+	@Path("/kc/updateGuids")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	@ReturnType("org.jboss.aerogear.unifiedpush.api.Alias")
+	public Response updateKcGuids(@Context HttpServletRequest request) {
+		try {
+			long updated = aliasService.updateKCUsersGuids();
+			return appendAllowOriginHeader(Response.ok(updated), request);
+		} catch (Exception e) {
+			logger.error("Cannot update guids", e);
+			return appendAllowOriginHeader(Response.status(Status.INTERNAL_SERVER_ERROR), request);
+		}
+	}
```

Comment:
If JSON payload is returned try to return an **object** or an **array/list** and not a scalar as it makes the return value clearer. In this case, perhaps `Response.ok(Collections.singletonMap("count", updated))`